pkg/image/external/external.go

```diff
@@ -90,8 +91,10 @@ func GetExternalImages(rancherVersion string, externalData map[string]interface{
 
 	for _, release := range compatibleReleases {
 		// Registries don't allow "+", so image names will have these substituted.
-		upgradeImage := fmt.Sprintf("rancher/%s-upgrade:%s", source, strings.Replace(release, "+", "-", -1))
+		upgradeImage := fmt.Sprintf("rancher/%s-upgrade:%s", source, strings.ReplaceAll(release, "+", "-"))
 		externalImagesMap[upgradeImage] = true
+		systemAgentInstallerImage := fmt.Sprintf("%s%s:%s", settings.SystemAgentInstallerImage.Default, source, strings.ReplaceAll(release, "+", "-"))
+		externalImagesMap[systemAgentInstallerImage] = true
```

Comment:
worth grabbing `SystemAgentUpgradeImage` too?

Reply:
I opened an issue https://github.com/rancher/rancher/issues/37596 to track additional changes in scripts/package, here's my explanation why https://github.com/rancher/rancher/pull/37586#discussion_r865510917

The issue with system-agent-upgrade-image is that it does not not have a default value in setting.go, nor does system-agent-version, which may introduce bugs when those aren't set. 

My decision was to ultimately keep the scope of this PR to a minimally required set of changes to keep the chance of a regression to a minimum.